bssl-compat/CMakeLists.txt

```diff
@@ -54,10 +50,15 @@ add_library(bssl-compat
     source/asn1_a_int.c
     )
 
-target_include_directories(bssl-compat PRIVATE
+target_link_libraries(bssl-compat PRIVATE crypto)
+add_dependencies(bssl-compat
+        OpenSSL)
+target_include_directories(bssl-compat PUBLIC
         # needed for ssl includes
-        ${openssl_SOURCE_DIR}/include
+        ${OPENSSL_INCLUDE_DIR}
         include
 #       needed only if ssl/ssl.h is used.
 #       ${openssl_SOURCE_DIR}
         )
+
+target_link_libraries(bssl-compat PRIVATE crypto)
```

Comment:
'crypto' is added twice, this second one should be for 'ssl'?
Speaking of which, I still get the OS defaults linked in:
$ ldd libbssl-compat.so 
	linux-vdso.so.1 (0x00007fff045ac000)
	libcrypto.so.1.1 => /lib/x86_64-linux-gnu/libcrypto.so.1.1 (0x00007fcf78de6000)
	libssl.so.1.1 => /lib/x86_64-linux-gnu/libssl.so.1.1 (0x00007fcf78d53000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fcf78b52000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fcf78b4d000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fcf78b48000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fcf79105000)
Is 'crypto' and 'ssl' referring to the system ones, should they be 'OpenSSL::Crypto' and 'OpenSSL:SSL'?

 If I blindly use those values I get:
CMake Error in CMakeLists.txt:
  Imported target "OpenSSL::Crypto" includes non-existent path
    "/home/pfl/projects/envoy-openssl/bssl-compat/build/openssl/include/openssl"

Reply:
The fix added solves this problem, so this part is solved.